### PR TITLE
Some more tests and samples

### PR DESCRIPTION
**Pull request by bdilly**

### CODE REVIEW

File: src/samples/flow/webservices/thingspeak-talkback-identify.fbp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#
+# There you should create a new Talkback. You need the talkback
+# id (not the name) and the API Key. Use these values as
+# options for talkback_add and talkback_execute nodes.
+#
+# You'll be able to see it throught the Thingspeak web interface as well,
+# if you want.
+
+
+# TODO wouldn't be better to accept API_KEY and ID through input ports?
```
Comment (cabelitos) [inline src/samples/flow/webservices/thingspeak-talkback-identify.fbp]: Are we leaving this TODO here?

File: src/samples/flow/webservices/thingspeak-talkback-smart-lock.fbp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Register at https://thingspeak.com or use another server (and
+# update node type options).
+#
+# There you should create a new Talkback. You need the talkback
+# id (not the name) and the API Key. Use these values as
+# options for talkback_add and talkback_execute nodes.
+
+
+# TODO wouldn't be better to accept API_KEY and ID through input ports?
```
Comment (cabelitos) [inline src/samples/flow/webservices/thingspeak-talkback-smart-lock.fbp]: Same thing

File: src/modules/flow/network/network.c [hunk 3]
```diff
@@ -190,7 +190,7 @@ network_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_
     SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
         SOL_FLOW_NODE_TYPE_NETWORK_BOOLEAN_OPTIONS_API_VERSION, -EINVAL);
 
-    if (!sol_network_subscribe_events(_on_network_event, mdata))
+    if (sol_network_subscribe_events(_on_network_event, mdata) < 0)
```
Comment (cabelitos) [inline src/modules/flow/network/network.c]: Should we use `SOL_INT_CHECK()` and properly return the error value

File: src/modules/linux-micro/network-up/network-up.c [hunk 4]
```diff
@@ -64,7 +64,7 @@ network_up_init(const struct sol_platform_linux_micro_module *module, const char
 {
     SOL_LOG_INTERNAL_INIT_ONCE;
 
-    if (!sol_network_subscribe_events(_network_event_cb, NULL))
+    if (sol_network_subscribe_events(_network_event_cb, NULL) < 0)
```
Comment (cabelitos) [inline src/modules/linux-micro/network-up/network-up.c]: Same thing

File: src/samples/network/network-status.c [hunk 5]
```diff
@@ -154,7 +154,7 @@ startup_network(void)
     if (!_compile_regex(regexp))
         goto err;
 
-    if (!sol_network_subscribe_events(_on_network_event, NULL))
+    if (sol_network_subscribe_events(_on_network_event, NULL) < 0)
```
Comment (cabelitos) [inline src/samples/network/network-status.c]: Same thing

File: src/samples/flow/webservices/thingspeak-talkback-identify.fbp (hunk 1 above)
Comment (bdilly) [inline src/samples/flow/webservices/thingspeak-talkback-identify.fbp]: I won't be able to modify the node type for now.
I'll open an issue later regarding accepting these info via input ports.
Tbh I'm still unsure about the best way of dealing with that, after all you don't want to let these values readable on memory. We should specify an approach and use it on all webservices node types - all of them need to deal with this issue.

File: src/modules/flow/network/network.c (hunk 3 above)
Comment (bdilly) [inline src/modules/flow/network/network.c]: good point

**Pull request merged**